Nodejs/Product/TestAdapter/TestFilesUpdateWatcher.cs

```diff
@@ -1,40 +1,33 @@
-// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
 
 using System;
 using System.Collections.Generic;
 using System.IO;
+using Microsoft.VisualStudio;
+using Microsoft.VisualStudio.Shell.Interop;
 using Microsoft.VisualStudio.TestPlatform.ObjectModel;
 
 namespace Microsoft.VisualStudioTools.TestAdapter
 {
-    internal class TestFilesUpdateWatcher : IDisposable
+    internal class TestFilesUpdateWatcher : IVsFileChangeEvents
     {
-        private readonly IDictionary<string, FileSystemWatcher> _fileWatchers;
+        private readonly IDictionary<string, uint> watchedFiles = new Dictionary<string, uint>(StringComparer.OrdinalIgnoreCase);
         public event EventHandler<TestFileChangedEventArgs> FileChangedEvent;
 
-        public TestFilesUpdateWatcher()
+        private readonly IVsFileChangeEx fileWatcher;
+
+        public TestFilesUpdateWatcher(IServiceProvider serviceProvider)
         {
-            _fileWatchers = new Dictionary<string, FileSystemWatcher>(StringComparer.OrdinalIgnoreCase);
+            this.fileWatcher = (IVsFileChangeEx)serviceProvider.GetService<SVsFileChangeEx>();
         }
 
         public bool AddWatch(string path)
         {
             ValidateArg.NotNull(path, "path");
-
-            if (!string.IsNullOrEmpty(path))
+            if (!this.watchedFiles.ContainsKey(path) && ErrorHandler.Succeeded(this.fileWatcher.AdviseFileChange(path, (uint)_VSFILECHANGEFLAGS.VSFILECHG_Add, this, out uint cookie)))
```

Comment:
Are you not interested in _VSFILECHANGEFLAGS.VSFILECHG_Del events?

Reply:
We care about all events.